proposed/http-client/http-client.md

```diff
@@ -7,43 +7,64 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
 "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
 interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
 
-## Specification
+## Goal
 
-### Client
+The goal of this PSR is to allow developers to create libraries decoupled from HTTP client
+implementations. This will make libraries more reusable as it reduces the number of
+dependencies and lowers the likelihood of version conflicts.
 
-An HTTP client has the responsibility to send a PSR-7 request and return a PSR-7
-response. Under the hood, the HTTP client MAY modify the request received from
-the user and/or the response received from the server. In this case, the request
-and the response MUST be consistent between the body and headers. For example, a
-server may return a gzip-encoded body and the client may decide to decode the
-body. If the client decodes the body, the client MUST also remove the
-`Content-Encoding` header and adjust the `Content-Length` header.
+A second goal is that HTTP clients can be replaced as per the
+[Liskov substitutions principle][Liskov]. This means that all clients MUST behave in the
+same way when sending a request.
 
-### Exceptions
+## Definitions
 
-All exceptions thrown by the client MUST implement `Psr\Http\Client\ClientExceptionInterface`.
+* **Client** - A Client is a library that implements this specification for the purposes of
+sending PSR-7-compatible HTTP Request messages and returning a PSR-7-compatible HTTP Response message to a Calling library.
+* **Calling Library** - A Calling Library is any code that makes use of a Client.  It does not implement
+this specification's interfaces but consumes an object that implements them (a Client).
 
-When the HTTP client is called with a request that is invalid and cannot be sent, the client
-MUST throw a `Psr\Http\Client\RequestExceptionInterface`. If there is an error
-with the network or the remote server cannot be reached, the HTTP client MUST throw
-a `Psr\Http\Client\NetworkExceptionInterface`.
+## Client
 
-Smaller issues that do not block the client from sending the request (such as
-invalid HTTP versions) MUST NOT result in exceptions.
+A Client is an object implementing `ClientInterface`.
 
-If the remote server answers with a response that can be parsed into a PSR-7 response,
-the client MUST NOT throw an exception. For example, response status codes in the
-400 and 500 range MUST NOT cause an exception.
+A Client MAY:
 
-## Goal
+* Elect to send an altered HTTP request from the one it was provided.  For example, it could
```

Comment:
Is "Elect" the correct word? It probably is but this word is only in my vocabulary if you talk about politicians. But that might only be me.

Reply:
not a native speaker, but my intuition says it is not wrong to say "elect" in this context. but "decide" may be easier to understand and would be correct too. wdyt @Crell ?

Reply:
"Elect" is technically valid English here, but it is rather formal.  I've changed it to "Choose".